### PR TITLE
[BE] enable UFMT for `torch/nn/modules`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128865
* #128593
* #128592
* __->__ #128594

Part of #123062

- #123062

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @albanD

### CODE REVIEW

**Review by mikaylagawarecki — APPROVED** [review]

**Comment by fbgheith** [comment]
Leads to test failures when run by Meta, will revert stack:

torchrec.modules.tests.test_lazy_extension.TestLazyModuleExtensionMixin
  test_source_code_parity_on_infer_parameters: AssertionError: '    [792 chars]rror(f\'module {self._get_name()} has not been[284 chars]me
' != '    [792 chars]rror(
                f"module {self._get_nam[310 chars]me
'

Failures:

  1) torchrec.modules.tests.test_lazy_extension.TestLazyModuleExtensionMixin: test_source_code_parity_on_infer_parameters
    1) AssertionError: '    [792 chars]rror(f\'module {self._get_name()} has not been[284 chars]me
' != '    [792 chars]rror(
                f"module {self._get_nam[310 chars]me
'
    Diff is 1699 characters long. Set self.maxDiff to None to see it. : Please make sure `LazyModuleExtensionMixin._infer_parameters` has the same source code as `LazyModuleMixin._infer_parameters` except the expected difference that is checked in this unit test.
      File "torchrec/modules/tests/test_lazy_extension.py", line 77, in test_source_code_parity_on_infer_parameters
        self.assertEqual(
                                                               --_ |""---__
Executed 1 example in 3.5s:                                 |'.|  ||  .    """|
  Successful: 0                                             | ||  || /|\""-.  |
  Failed: 1                                                 | ||  ||  |    |  |
  Skipped: 0                                                | ||  ||  |   \|/ |
  Not executed: 5                                           |."|  ||  --"" '__|
https://testslide.readthedocs.io/                              --" |__---"""

File: torch/nn/modules/lazy.py [hunk 1]
```diff
@@ -256,8 +280,10 @@ def _infer_parameters(self: _LazyProtocol, module, args, kwargs=None):
         delattr(module, '_load_hook')
         if module.cls_to_become is not None:
             module.__class__ = module.cls_to_become
-
+    # fmt: on
```
Comment (XuehaiPan) [inline torch/nn/modules/lazy.py:283]: `torchrec` requires this to be exactly the same as the one in `torchrec.modules.lazy_extension.LazyModuleExtensionMixin._infer_parameters`

https://github.com/pytorch/torchrec/blob/eca606d1ccb66ccec823254675a1eff5853270b1/torchrec/modules/lazy_extension.py#L159-L182

https://github.com/pytorch/torchrec/blob/eca606d1ccb66ccec823254675a1eff5853270b1/torchrec/modules/tests/test_lazy_extension.py#L57-L83

**Pull request closed without merging**